tests/testthat/test-vector_logic_linter.R

```diff
@@ -0,0 +1,81 @@
+test_that("vector_logic_linter skips allowed usages", {
+  expect_lint("if (TRUE) 5 else if (TRUE) 2", NULL, vector_logic_linter())
```

Comment:
BTW: Do we have a linter for this kind of unreachable code?

Reply:
I have `unreachable_code_linter` but the logic is limited so far. we can keep extending that.